nlp4all/models.py

```diff
@@ -322,18 +322,30 @@ class User(db.Model, UserMixin):
 
     def get_reset_token(self, expires_sec=1800):
         """Get a reset token."""
-        jws = Serializer(app.config["SECRET_KEY"], expires_sec)
-        return jws.dumps({"user_id": self.id}).decode("utf-8")
+        reset_token = jwt.encode(
+            {
+                "user_id": self.id,
+                "exp": datetime.now(tz=timezone.utc)
+                        + timedelta(seconds=expires_sec)
+            },
+            app.config['SECRET_KEY'],
+            algorithm="HS256"
+        )
+        return reset_token
 
     @staticmethod
     def verify_reset_token(token):
-        """Verify a reset token."""
-        jws = Serializer(app.config["SECRET_KEY"])
         try:
-            user_id = jws.loads(token)["user_id"]
-        except SignatureExpired:
-            return None
-        except ValueError:
+            data = jwt.decode(
+                token,
+                app.config['SECRET_KEY'],
+                leeway=timedelta(seconds=10),
+                algorithms=["HS256"]
+                )
+
+            user_id = data['user_id']
+
+        except:
```

Comment:
this `except` is a bit too broad, are there any other exceptions expected? we can limit the scope by catching expected exceptions

Reply:
[only two reasonable errors to be expected(expired and invalid).
](https://pyjwt.readthedocs.io/en/stable/api.html#exceptions)
They are kinda handle where the function is called:
```python
 @app.route("/reset_password/<token>", methods=["GET", "POST"])

def reset_token(token):
    """Reset password token page"""
    if current_user.is_authenticated:
        return redirect(url_for("home"))
    user = User.verify_reset_token(token)
    if user is None:
        flash("That is an invalid or expired token", "warning")
        return redirect(url_for("reset_request"))
```

Tell me what you think :)

Reply:
I could handle `InvalidSignatureError ` and `ExpiredSignatureError ` independently, but at the end of the day the user will have to try again.